api/api.go

```diff
@@ -13,21 +14,86 @@ type API struct {
 	dataStore DataStore
 }
 
+// Setup function sets up the api with authentication and returns an api
+func Setup(ctx context.Context, router *mux.Router, dataStore DataStore) *API {
+	api := &API{
+		Router:    router,
+		dataStore: dataStore,
+	}
+
+	api.get(
+		"/v1/cache-times/{id}",
+		func(w http.ResponseWriter, req *http.Request) { api.GetCacheTime(ctx, w, req) },
+	)
+
+	api.put(
+		"/v1/cache-times/{id}",
+		api.isAuthenticated(func(w http.ResponseWriter, req *http.Request) { api.CreateOrUpdateCacheTime(ctx, w, req) }),
+	)
+
+	api.post(
+		"/mongocheck",
+		api.isAuthenticated(
+			func(w http.ResponseWriter, req *http.Request) { api.AddDataSets(ctx) }),
+	)
+
+	api.get(
+		"/mongocheck",
+		api.isAuthenticated(
+			func(w http.ResponseWriter, req *http.Request) { api.GetDataSets(ctx) }),
+	)
+
+	return api
+}
+
 // Setup function sets up the api and returns an api
-func Setup(ctx context.Context, r *mux.Router, dataStore DataStore) *API {
+func SetupNoAuth(ctx context.Context, router *mux.Router, dataStore DataStore) *API {
```

Comment:
Is the purpose of this function just for stubbing auth in the tests? If so, perhaps it should live in api_test.

Reply:
Yes agree. I will move it into api_test